### PR TITLE
qemu_vm: Add '-numa cpu' to qemu command line

### PR DESCRIPTION
**Pull request by yanan-fu**
id: 1859025

Signed-off-by: Yanan Fu <yfu@redhat.com>

### CODE REVIEW

**Comment by yanan-fu** [comment]
@PaulYuuu  Could you help review, thanks a lot!

File: virttest/qemu_vm.py [hunk 1]
```diff
@@ -1623,6 +1642,18 @@ def sort_key(dev):
...
+            numa_cpu_params = params.object_params(numa_cpu)
+            numa_cpu_nodeid = numa_cpu_params.get("numa_cpu_nodeid")
+            numa_cpu_socketid = numa_cpu_params.get("numa_cpu_socketid")
+            numa_cpu_dieid = numa_cpu_params.get("numa_cpu_dieid")
+            numa_cpu_coreid = numa_cpu_params.get("numa_cpu_coreid")
+            numa_cpu_threadid = numa_cpu_params.get("numa_cpu_threadid")
+            cmdline = add_numa_cpu(devices, numa_cpu_nodeid, numa_cpu_socketid,
+                                   numa_cpu_dieid, numa_cpu_coreid,
+                                   numa_cpu_threadid)
+            devices.insert(StrDev('numa', cmdline=cmdline))
```
Comment (PaulYuuu) [inline virttest/qemu_vm.py:1655]: I think better to assign a different ID to each node to distinguish them. Not only here, but also `# Add numa nodes` above.

Comment (yanan-fu) [reply virttest/qemu_vm.py:1656]: > I think better to assign a different ID to each node to distinguish them. Not only here, but also `# Add numa nodes` above.
Yes, i should distinguish numa cpu and numa node, but assign different ID for each node is not very reasonable i think.
It is a type as my understanding:
https://github.com/avocado-framework/avocado-vt/blob/db232b71a3feaaab7737c8b02cf450acf4301b78/virttest/qemu_devices/qdevices.py#L348

File: virttest/qemu_vm.py [hunk 2]
```diff
@@ -1295,6 +1295,25 @@ def add_numa_node(devices, memdev=None, mem=None,
                 numa_cmd += ",nodeid=%s" % nodeid
             return numa_cmd
 
+        def add_numa_cpu(devices, nodeid, socketid=None, dieid=None,
+                         coreid=None, threadid=None):
+            """
+            This function is used to add numa cpu to guest command line
+            node-id=node[,socket-id=x][,core-id=y][,thread-id=z]
```
[the remark this answers is not in the document]
Comment (yanan-fu) [reply virttest/qemu_vm.py]: Updated, thanks ~

**Review by PaulYuuu — APPROVED** [review]

**Comment by PaulYuuu** [comment]
LGTM

File: virttest/qemu_vm.py [hunk 3]
```diff
@@ -1296,6 +1296,25 @@ def add_numa_node(devices, memdev=None, mem=None,
                 numa_cmd += ",nodeid=%s" % nodeid
             return numa_cmd
 
+        def add_numa_cpu(devices, nodeid, socketid=None, dieid=None,
```
Comment (vivianQizhu) [inline virttest/qemu_vm.py:1299]: UnderScoreCase name please, `node_id`.

File: virttest/qemu_vm.py [hunk 4]
```diff
@@ -1296,6 +1296,25 @@ def add_numa_node(devices, memdev=None, mem=None,
...
+                return ""
+            numa_cpu_cmd = " -numa cpu,node-id=%s" % nodeid
+            if socketid is not None:
+                numa_cpu_cmd += ",socket-id=%s" % socketid
+            if dieid is not None:
+                numa_cpu_cmd += ",die-id=%s" % dieid
+            if coreid is not None:
+                numa_cpu_cmd += ",core-id=%s" % coreid
+            if threadid is not None:
+                numa_cpu_cmd += ",thread-id=%s" % threadid
```
Comment (vivianQizhu) [inline virttest/qemu_vm.py:1315]: What about:
```
for name, value in {"socket-id": socket_id, "die-id": die_id, "core-id": core_id, "thread-id": thread_id}
    if value is not None:
        numa_cpu_cmd += ",%s=%s" % (name, value)
```

Comment (yanan-fu) [reply virttest/qemu_vm.py]: Updated, thanks~

**Comment by vivianQizhu** [comment]
Please add doc string in base.cfg for the new parameters, thanks.

**Comment by yanan-fu** [comment]
> Please add doc string in base.cfg for the new parameters, thanks.

Added, thanks!

**Review by vivianQizhu — APPROVED** [review]

**Pull request merged**